### ISSUE TITLE
3D demo: Warnings about ARAP in the Surface_mesh_parameterization plugin

### ISSUE THREAD

**Issue by lrineau** [CGAL/cgal#2672]
## Issue Details

Warnings on Windows about ARAP and the Surface_mesh_parameterization plugin:
```
test\Polyhedron_Demo\Plugins\Surface_mesh\Parameterization_plugin.cpp(320): warning C4267: 'return': conversion from 'size_t' to 'int', possible loss of data
test\Polyhedron_Demo\Plugins\Surface_mesh\Parameterization_plugin.cpp(751): warning C4267: 'argument': conversion from 'size_t' to 'const unsigned int', possible loss of data
C:\CGAL\test\CGAL-4.12-Ic-113\include\CGAL/Surface_mesh_parameterization/ARAP_parameterizer_3.h(569): warning C4267: 'return': conversion from 'size_t' to 'int', possible loss of data
C:\CGAL\test\CGAL-4.12-Ic-113\include\CGAL/Surface_mesh_parameterization/ARAP_parameterizer_3.h(556): note: while compiling class template member function 'int CGAL::Surface_mesh_parameterization::ARAP_parameterizer_3<Seam_mesh,CGAL::Default,CGAL::Default>::solve_cubic_equation(const double,const double,const double,const double,std::vector<double,std::allocator<_Ty>> &) const'
        with
        [
            _Ty=double
        ]
C:\CGAL\test\CGAL-4.12-Ic-113\include\CGAL/Surface_mesh_parameterization/ARAP_parameterizer_3.h(804): note: see reference to function template instantiation 'int CGAL::Surface_mesh_parameterization::ARAP_parameterizer_3<Seam_mesh,CGAL::Default,CGAL::Default>::solve_cubic_equation(const double,const double,const double,const double,std::vector<double,std::allocator<_Ty>> &) const' being compiled
        with
        [
            _Ty=double
        ]
test\Polyhedron_Demo\Plugins\Surface_mesh\Parameterization_plugin.cpp(902): note: see reference to class template instantiation 'CGAL::Surface_mesh_parameterization::ARAP_parameterizer_3<Seam_mesh,CGAL::Default,CGAL::Default>' being compiled
```
https://cgal.geometryfactory.com/CGAL/testsuite/results-4.12-Ic-113.shtml#Polyhedron_Demo

They are **only in `integration`**, but I do not understand which PR is the cause of the warnings. To @maxGimeno, @afabri, @MaelRL: does one of you knows?

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Windows, all versions
* Compiler: MSVC
* Release or debug mode: both
* Specific flags used (if any):
* CGAL version:  ??

**Comment by MaelRL** [comment]
It's from ARAP's code, just some functions returning a `.size()` and declaring an `int` return type.

Somewhat surprising that it wasn't detected locally or when ARAP was first integrated.

I'll fix it in the Core branch (https://github.com/CGAL/cgal/pull/2652).

**Comment by lrineau** [comment]
Fixed by #2652, in CGAL-4.11.1. Mael, can you check it is correctly fixed?

**Comment by MaelRL** [comment]
All SMP-related warnings on Windows seem to be gone from the Polyhedron demo.
